api/proto/teleport/common/v1/common.proto

```diff
@@ -0,0 +1,59 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package teleport.common.v1;
```

Comment:
I'm not sure a "common" package is a good idea, chances are it'll inflate over time.

Reply:
Do you have a suggestion? We do have a number of common needs -- `Traits`, `ResourceHeader`, and `Metadata` will be seen pretty commonly.

Reply:
Talking on slack, used `header` and `trait` packages separately.